frontend/service/Ethers.ts

```diff
@@ -89,6 +90,65 @@ const checkRpc = async (rpc: string): Promise<boolean> => {
   }
 };
 
+const BACK_TRACK_BLOCKS = 9000;
+const MAX_ROUNDS = 10;
+
+/**
+ * Get the latest transaction details for the given contract address
+ */
+export const getLatestTransaction = async (
+  rpc: string,
+  contractAddress: Address,
+): Promise<TransactionInfo | null> => {
+  const provider = new providers.StaticJsonRpcProvider(rpc, {
+    name: 'Gnosis',
+    chainId: 100, // we currently only support Gnosis Trader agent
+  });
+
+  const latestBlock = await provider.getBlockNumber();
+
+  const getLogsList = async (
```

Comment:
can we pull this into out of this function? may be useful elsewhere in function, and no need to recreate on every call